src/services/review/RepoCheckerService.ts

```diff
@@ -64,6 +69,65 @@ export default class RepoCheckerService {
     this.repoRepository = repoRepository
   }
 
+  isCurrentlyLocked(repo: string): ResultAsync<boolean, never> {
+    const logsFilePath = path.join(SITE_CHECKER_REPO_PATH, repo)
+    // create logs folder if it does not exist
+
+    if (!fs.existsSync(logsFilePath)) {
+      fs.mkdirSync(logsFilePath, { recursive: true })
+    }
+
+    // check if checker.lock exists
+    const lockFilePath = path.join(logsFilePath, "checker.lock")
+    return ResultAsync.fromPromise(
+      fs.promises.access(lockFilePath, fs.constants.F_OK),
+      () => okAsync(false)
+    )
+      .map(() => true)
+      .orElse(() => ok(false))
+  }
+
+  createLock(repo: string): ResultAsync<undefined, SiteCheckerError> {
+    // create a checker.lock file
+    const lockFilePath = path.join(SITE_CHECKER_REPO_PATH, repo, "checker.lock")
+    return ResultAsync.fromPromise(
+      fs.promises.writeFile(lockFilePath, "locked"),
+      (error) => new SiteCheckerError(`${error}`)
+    )
+      .andThen(() =>
+        ResultAsync.fromPromise(
+          this.git
+            .cwd({ path: SITE_CHECKER_REPO_PATH, root: false })
+            .add([`${path.join(repo, "checker.lock")}`])
+            .commit(`Create lock for ${repo}`)
+            .push(),
+          (error) => new SiteCheckerError(`${error}`)
+        )
+      )
+      .map(() => undefined)
+  }
+
+  deleteLock(repo: string): ResultAsync<undefined, SiteCheckerError> {
+    // delete the checker.lock file
+    const lockFilePath = path.join(SITE_CHECKER_REPO_PATH, repo, "checker.lock")
+    return ResultAsync.fromPromise(
+      fs.promises.unlink(lockFilePath),
+      (error) => new SiteCheckerError(`${error}`)
+    )
+      .map(() => undefined)
+      .andThen(() =>
+        ResultAsync.fromPromise(
+          this.git
+            .cwd({ path: SITE_CHECKER_REPO_PATH, root: false })
+            .add([`${path.join(repo, "checker.lock")}`])
+            .commit(`Delete lock for ${repo}`)
+            .push(),
+          (error) => new SiteCheckerError(`${error}`)
+        )
+      )
+      .map(() => undefined)
```

Comment:
same here

Reply:
i think is this weird quirk of neverthrow, types it as void and then void and undefined conflict

Reply:
not entirely sure - i think we can fix this just by hinting the return type as `void` above adn omitting this step. i think this is a quick fix that eliminates uncertainty in our codebase, which is why i pointed it out!

Reply:
good point. am changing this to return the push result instead, rather than void